base/shell/explorer/trayprop.cpp

```diff
@@ -21,441 +21,240 @@
-                                (LPARAM)pPropInfo->hTaskbarBitmap);
-                }
-            }
-        }
-        
-        if (bHideInactive)
-        {
-            EnableWindow(hwndCustomizeNotifyButton, TRUE);
-            if (bShowClock)
-            {
-                EnableWindow(hwndSeconds, TRUE);
-                if (bShowSeconds)
-                    lpTrayImageName = MAKEINTRESOURCEW(IDB_SYSTRAYPROP_HIDE_SECONDS);
-                else
-                    lpTrayImageName = MAKEINTRESOURCEW(IDB_SYSTRAYPROP_HIDE_CLOCK);
-            }
-            else
-            {
-                SendMessage(hwndSeconds, BM_SETCHECK, BST_UNCHECKED, 0);
-                EnableWindow(hwndSeconds, FALSE);
-                lpTrayImageName = MAKEINTRESOURCEW(IDB_SYSTRAYPROP_HIDE_NOCLOCK);
-            }
-        }
-        else
-        {
-            EnableWindow(hwndCustomizeNotifyButton, FALSE);
-            if (bShowClock)
-            {
-                EnableWindow(hwndSeconds, TRUE);
-                if (bShowSeconds)
-                    lpTrayImageName = MAKEINTRESOURCEW(IDB_SYSTRAYPROP_SHOW_SECONDS);
-                else
-                    lpTrayImageName = MAKEINTRESOURCEW(IDB_SYSTRAYPROP_SHOW_CLOCK);
-            }
-            else
-            {
-                SendMessage(hwndSeconds, BM_SETCHECK, BST_UNCHECKED, 0);
-                EnableWindow(hwndSeconds, FALSE);
-                lpTrayImageName = MAKEINTRESOURCEW(IDB_SYSTRAYPROP_SHOW_NOCLOCK);
-            }
-        }
-        
-        if (lpTrayImageName)
-        {
-            if (pPropInfo->hTrayBitmap)
-            {
-                DeleteObject(pPropInfo->hTrayBitmap);
-            }
-
-            pPropInfo->hTrayBitmap = (HBITMAP)LoadImageW(hExplorerInstance,
-                                                  lpTrayImageName,
-                                                  IMAGE_BITMAP,
-                                                  0,
-                                                  0,
-                                                  LR_DEFAULTCOLOR);
-            if (pPropInfo->hTrayBitmap)
-            {
-                hwndTrayBitmap = GetDlgItem(pPropInfo->hTaskbarWnd,
-                                        IDC_TASKBARPROP_NOTIFICATIONBITMAP);
-                if (hwndTrayBitmap)
-                {
-                    BITMAP bm;
-                    GetObject(pPropInfo->hTrayBitmap, sizeof(bm), &bm);
-                    SetWindowPos(hwndTrayBitmap, NULL, 0, 0, bm.bmWidth + 2, bm.bmHeight + 2,
-                                 SWP_NOACTIVATE | SWP_NOMOVE | SWP_NOZORDER);
-                    SendMessage(hwndTrayBitmap,
-                                STM_SETIMAGE,
-                                IMAGE_BITMAP,
-                                (LPARAM)pPropInfo->hTrayBitmap);
-                }
-            }
-        }
+            uImageId = IDB_TASKBARPROP_NOLOCK_GROUP_QL;
+
+        if (uImageId)
```

Comment:
This check is unnecessary, the if above covers all possible cases. An `else ASSERT(FALSE)` above may or may not make sense.